src/SDK/Resource/ResourceInfo.php

```diff
@@ -51,17 +56,57 @@ public static function merge(ResourceInfo ...$resources): self
 
     public static function defaultResource(): self
     {
-        return (new Detectors\Composite([
-            new Detectors\Environment(),
+        $detectors = (new ResourceInfo(new Attributes()))->getListFromEnvironment(Env::OTEL_PHP_DETECTORS);
```

Comment:
Could you use `Accessor::getList(Env::OTEL_PHP_DETECTORS)` here?
This method and Env Vars in general are static anyway, so not creating objects would be a bit more performant.